### DISCUSSION TITLE
Ethernet based update capability?

### DISCUSSION THREAD

**Discussion by ccrome** [adafruit/tinyuf2#296] [Ideas]
Is there any desire/openness/thoughts about making tinyuf2 work over Ethernet for Ethernet based embedded devices?

I know this is somewhat orthogonal to the the tinyusb heritage of tinyuf2, but we have some Ethernet only embedded systems that need ota updating and it would be great to use the tinyuf2 framework for doing the updates. 

I've used uTasker with lwip for this before.  I don't really know what other alternatives there are out there.  The esphome updater is pretty excellent, but I don't know if it's general purpose firmware update system:  I kind of doubt it. 

If you had to do an Ethernet only update today, how would you go about it?

Thanks!

**Comment by tannewt** [comment]
I don't think we'd want this in TinyUF2. Adafruit supports development of TinyUF2 and wouldn't use Ethernet updating.
